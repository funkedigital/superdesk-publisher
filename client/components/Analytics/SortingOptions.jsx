import React from "react";

import Dropdown from "../UI/Dropdown";

const SortingOptions = ({ filters, setFilters }) => {
  const setSort = value => {
    filters.sort = value;
    setFilters(filters);
  };

  const setOrder = value => {
    filters.order = value;
    setFilters(filters);
  };

  return (
    <React.Fragment>
      <Dropdown
        button={
          <button className="dropdown__toggle">
            {filters.sort === "published_at" ? "Publish date" : "Page views"}
            <span className="dropdown__caret"></span>
          </button>
        }
      >
        <li>
          <button onClick={() => setSort("published_at")}>Publish date</button>
        </li>
        <li>
          <button
            onClick={() => setSort("article_statistics.page_views_number")}
          >
            Page views
          </button>
        </li>
      </Dropdown>
      {filters.order === "desc" ? (
        <a
          className="icn-btn"
          sd-tooltip="Descending"
          flow="bottom"
          onClick={() => setOrder("asc")}
        >
          <i className="icon-descending"></i>
        </a>
      ) : (
        <a
          className="icn-btn"
          sd-tooltip="Ascending"
          flow="bottom"
          onClick={() => setOrder("desc")}
        >
          <i className="icon-ascending"></i>
        </a>
      )}
    </React.Fragment>
  );
};

export default SortingOptions;
