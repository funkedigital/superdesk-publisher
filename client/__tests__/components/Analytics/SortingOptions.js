import React from "react";
import SortingOptions from "../../../components/Analytics/SortingOptions";
import { render } from "@testing-library/react";

describe("Analytics/SortingOptions", () => {
  const filters = {
    sort: "publish_date",
    order: "desc"
  };

  it("renders correctly", () => {
    const { container } = render(<SortingOptions filters={filters} setFilters={jest.fn()} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
