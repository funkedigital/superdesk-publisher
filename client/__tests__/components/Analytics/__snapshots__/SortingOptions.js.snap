// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics/SortingOptions renders correctly 1`] = `
<div
  class="dropdown dropdown--align-right"
  flow="left"
>
  <span>
    <button
      class="dropdown__toggle"
    >
      Page views
      <span
        class="dropdown__caret"
      />
    </button>
  </span>
  <ul
    class="dropdown__menu dropdown--align-right"
  >
    <li>
      <button>
        Publish date
      </button>
    </li>
    <li>
      <button>
        Page views
      </button>
    </li>
  </ul>
</div>
`;
